trae_agent/utils/config.py

```diff
@@ -1,210 +1,342 @@
+            lakeview_model_name = lakeview.get("model", None)
+            if lakeview_model_name is None:
+                raise ConfigError("No model provided for lakeview")
+            lakeview_model = config_models[lakeview_model_name]
+            config.lakeview = LakeviewConfig(
+                model=lakeview_model,
+            )
+        else:
+            config.lakeview = None
+
+        # Parse agents
+        agents = yaml_config.get("agents", None)
+        if agents is not None and len(agents.keys()) > 0:
+            for agent_name, agent_config in agents.items():
+                agent_model_name = agent_config.get("model", None)
+                if agent_model_name is None:
+                    raise ConfigError(f"No model provided for {agent_name}")
+                try:
+                    agent_model = config_models[agent_model_name]
+                except KeyError as e:
+                    raise ConfigError(f"Model {agent_model_name} not found") from e
+                match agent_name:
+                    case "trae_agent":
+                        trae_agent_config = TraeAgentConfig(**agent_config)
+                        trae_agent_config.model = agent_model
+                        if trae_agent_config.enable_lakeview and config.lakeview is None:
+                            raise ConfigError("Lakeview is enabled but no lakeview config provided")
+                        config.trae_agent = trae_agent_config
+                    case _:
+                        raise ConfigError(f"Unknown agent: {agent_name}")
+        else:
+            raise ConfigError("No agent configs provided")
+        return config
+
+    def resolve_config_values(
+        self,
+        *,
+        model_providers: dict[str, ModelProvider] | None = None,
+        provider: str | None = None,
+        model: str | None = None,
+        model_base_url: str | None = None,
+        api_key: str | None = None,
+        max_steps: int | None = None,
+    ):
+        if self.trae_agent:
+            self.trae_agent.resolve_config_values(
+                max_steps=max_steps,
+            )
+            self.trae_agent.model.resolve_config_values(
+                model_providers=model_providers,
+                provider=provider,
+                model=model,
+                model_base_url=model_base_url,
+                api_key=api_key,
+            )
+        return self
+
+    @classmethod
+    def create_from_legacy_config(
+        cls,
+        *,
+        legacy_config: LegacyConfig | None = None,
+        config_file: str | None = None,
+    ) -> "Config":
+        if legacy_config and config_file:
+            raise ConfigError("Only one of legacy_config or config_file should be provided")
+
+        if config_file:
+            legacy_config = LegacyConfig(config_file)
+        elif not legacy_config:
+            raise ConfigError("No legacy_config or config_file provided")
+
+        model_provider = ModelProvider(
```

Comment:
Hear the code maybe a bit too long ? I guess create a handler call which takes legacy_config as the parameter. Here's what I think would be proper
```suggestion
        def _model_provider_legacy_handler(legacy_config) -> ModelProvider:
```